src/app/switcher/switcher-program/switcher-program.component.ts

```diff
@@ -23,15 +24,23 @@ import { UtilsService } from '@services/utils.service';
 export class SwitcherProgramComponent extends RouterEnter implements AfterContentChecked {
   routeUrl = '/switcher/switcher-program';
   programs: Array<ProgramObj>;
+  stacks: Stack[];
+
   constructor(
     public loadingController: LoadingController,
     public router: Router,
     private pusherService: PusherService,
     private switcherService: SwitcherService,
     private newRelic: NewRelicService,
     private notificationService: NotificationService,
-    private utils: UtilsService
-  ) { super(router); }
+    private utils: UtilsService,
+    private activatedRoute: ActivatedRoute,
+  ) {
+    super(router);
+    activatedRoute.data.subscribe(data => {
```

Comment:
We don't need to get the stacks from the resolver, we will get the stacks from the local storage anyway.

Reply:
hi @shawnm0705 thanks for your suggestion, but both resolver and localstorage serve the same purpose.
Resolver has better advantages where the localStorage fallshort. Resolver makes sure "stacks" get pulled correctly before user get redirected to the page, so switcher page is always guaranteed with "stacks" value to started with.

Unless the speed of one single API call to `https://login.practera.com/stacks` concerns you, otherwise we shouldn't be too worried about the loading speed, anything other than this `https://login.practera.com/stacks` API call is still happen inside our switcher loading screen.